### PR TITLE
Added FAQ for developers

### PR DESCRIPTION
**Pull request by maudnals**

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+expected to consume some or even a large part of their budget, because they need to access some
+information about the agent to guarantee a great user experience or use specific features. You can
+think of it like an actual budget: it belongs to your site, and you as a site developer will decide
+how to allocate it. Limitations will kick in only if you spend all of your site's budget, because
+this will mean that your site is accessing a combination of APIs that can be used to uniquely
+identify users. Typically:
+
+- Sites that covertly track users will hit their budget, and they'll need to change.
+- Some applications, such as 3D games or video conferencing, may not be able to run within any
+  reasonable privacy budget. For such applications, there will need to be an "escape valve": a
```
Comment (bslassey) [inline faq.md:50]:
```suggestion
  reasonable privacy budget. For such applications, there may need to be an "escape valve": a
```

File: faq.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+First, note that **consuming some of your site's budget is not a problem**. In many cases, sites are
+expected to consume some or even a large part of their budget, because they need to access some
+information about the agent to guarantee a great user experience or use specific features. You can
+think of it like an actual budget: it belongs to your site, and you as a site developer will decide
+how to allocate it.
+
+How much budget specific APIs will consume isn't defined yet. Many APIs are expected to consume at
+least some of the privacy budget, but the handful of APIs that are **likely** to consume larger
+amounts of the privacy budget are APIs that access browser settings, APIs that enable to read the
```
Comment (bslassey) [inline faq.md:28]:
```suggestion
amounts of the privacy budget are APIs that access browser attribute, APIs that enable to read the
```

File: faq.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+expected to consume some or even a large part of their budget, because they need to access some
+information about the agent to guarantee a great user experience or use specific features. You can
+think of it like an actual budget: it belongs to your site, and you as a site developer will decide
+how to allocate it. Limitations will kick in only if you spend all of your site's budget, because
+this will mean that your site is accessing a combination of APIs that can be used to uniquely
+identify users. Typically:
+
+- Sites that covertly track users will hit their budget, and they'll need to change.
+- Some applications, such as 3D games or video conferencing, may not be able to run within any
+  reasonable privacy budget. For such applications, there will need to be an "escape valve": a
```
Comment (bslassey) [inline faq.md]:
```suggestion
  reasonable privacy budget. For such applications, there may need to be an "escape valve": a
```

File: faq.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+This is an open question, and it will need to be publicly established in the future. It depends on
+the approach different browser vendors decide to take.
+
+### Is the privacy budget feasible?
+
+This proposal is at an early stage and exact privacy budget limits are to be determined. But there
+are reasons to think that a privacy budget is feasible: it takes about **32 bits of entropy** to
+uniquely identify people on the web, but mobile web users (which make up most of web users) can only
+expose at most about **13 bits** of entropy.
```
Comment (bslassey) [inline faq.md:135]:
```suggestion
uniquely identify people on the web.
```

File: faq.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+- There are about [4.6 billion of web
+  users](https://www.statista.com/statistics/617136/digital-population-worldwide/). So it takes
+  log2(4.6 billion) = about 32 bits of entropy to uniquely identify people on the web.
+- Mobile web users make up [most of web
+  users](https://gs.statcounter.com/platform-market-share/desktop-mobile-tablet/).
+- There's an order of 10,000 different mobile device models (estimates range from 4,000 to 60,000+).
+  Devices of the same model have the same fingerprint. So mobile web users can only expose at most
+  log2(10,000) = about 13 bits of entropy. Note that the mobile device population is very unevenly
+  distributed over the thousands of device models, so the entropy of knowing a mobile device model
+  may be significantly lower than 13 bits.
```
Comment (bslassey) [inline faq.md:148]:
```suggestion
```

Comment (bslassey) [reply faq.md:148]: This is a rough estimate of how much entropy the device model exposes, although as it notes it is not an even distribution. The translation into "but mobile web users can only expose 13 bits" isn't right. **If** a site requests the device model and device model were to be measured as exposing 13 bits (which seems like an over estimate, but leave that aside for now) then those 13 bits would be deducted from the budget. So, if the budget were, say, 28 bits there would be 15 bits remaining in the budget, not 13.

**Review by bslassey — APPROVED** [review]

File: faq.md (hunk 3 above)
Comment (maudnals) [reply faq.md]: LGTM

**Pull request merged**